core/src/main/java/io/questdb/griffin/engine/table/ShowPartitionsRecordCursorFactory.java

```diff
@@ -0,0 +1,434 @@
+                            path.parent().concat(TableUtils.TXN_FILE_NAME).$();
+                            if (ff.exists(path)) {
+                                if (detachedTxReader == null) {
+                                    detachedTxReader = new TxReader(FilesFacadeImpl.INSTANCE);
+                                }
+                                detachedTxReader.ofRO(path, partitionBy);
+                                detachedTxReader.unsafeLoadAll();
+                                long timestamp = PartitionBy.parsePartitionDirName(partitionName, partitionBy);
+                                int pIndex = detachedTxReader.getPartitionIndex(timestamp);
+                                numRows = detachedTxReader.getPartitionSize(pIndex);
+                                if (PartitionBy.isPartitioned(partitionBy) && numRows > 0L) {
+                                    int tsIndex = detachedMetaReader.getTimestampIndex();
+                                    dynamicTsColName = detachedMetaReader.getColumnName(tsIndex);
+                                }
+                                // could set dynamicPartitionIndex to -pIndex
+                            } else {
+                                LOG.error().$("detached partition does not have meta file [path=").$(path).I$();
+                            }
+                        } else {
+                            LOG.error().$("detached partition meta does not match [path=").$(path).I$();
+                        }
+                    } else {
+                        LOG.error().$("detached partition does not have meta file [path=").$(path).I$();
+                    }
+                    path.parent();
+                }
+                partitionSize = ff.getDirectoryContentSize(path.$());
+                if (PartitionBy.isPartitioned(partitionBy) && numRows > 0L) {
+                    TableUtils.dFile(path.slash$(), dynamicTsColName, TableUtils.COLUMN_NAME_TXN_NONE);
+                    int fd = -1;
+                    try {
+                        fd = TableUtils.openRO(ff, path, LOG);
+                        long lastOffset = (numRows - 1) * ColumnType.sizeOf(ColumnType.TIMESTAMP);
+                        minTimestamp = ff.readNonNegativeLong(fd, 0);
+                        maxTimestamp = ff.readNonNegativeLong(fd, lastOffset);
+                    } catch (CairoException e) {
+                        if (partitionIndex < partitionCount) {
+                            throw CairoException.critical(ff.errno()).put("no file found for designated timestamp column [path=").put(path).put(']');
+                        }
+                        dynamicPartitionIndex = Integer.MIN_VALUE;
+                        LOG.error().$("no file found for designated timestamp column [path=").$(path).I$();
+                    } finally {
+                        ff.close(fd);
```

Comment:
there is no test that fails to open file, if this were to happen we will have double-whammy trying to close -1 fd

Reply:
Files.close:

```
public static int close(int fd) {
        // do not close `stdin` and `stdout`
        if (fd > 1) {
            assert auditClose(fd);
            int res = close0(fd);
            if (res == 0) {
                OPEN_FILE_COUNT.decrementAndGet();
            }
            return res;
        }
        // failed to close
        return -1;
    }
```